packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { TimeConstants, addMinutes, formatTimeString, ceilMinuteToIncrement } from '@fluentui/date-time-utilities';
+import { ComboBox, IComboBox, IComboBoxOption } from '../../ComboBox';
+import { ITimePickerProps, ITimeRange } from './TimePicker.types';
+
+// Valid KeyChars for user input
+const KEYCHAR_0 = 48;
+const KEYCHAR_COLON = 58;
+const KEYCHAR_SPACE = 32;
+const KEYCHAR_A = 65;
+const KEYCHAR_M = 77;
+const KEYCHAR_P = 80;
+
+export const TimePicker = ({
+  label,
+  increments = 30,
+  showSeconds = false,
+  allowFreeform = true,
+  useHour12 = false,
+  timeRange = { start: -1, end: -1 },
+  onFormatDate,
+  onValidateUserInput,
+  ...rest
+}: ITimePickerProps) => {
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>(0);
+  const [userText, setUserText] = React.useState<string>('');
+  const [errorMessage, setErrorMessage] = React.useState<string>('');
+
+  const defaultTime = generateDefaultTime(increments, timeRange);
+  const optionsCount = getDropdownOptionsCount(increments, timeRange);
+  const timePickerOptions: IComboBoxOption[] = Array(optionsCount)
```

Comment:
This must be wrapped in `React.useMemo`, or the perf will be terrible since the options will be re-created every render.

Also, `Array.fill` doesn't work in IE 11, so you'll need to use a different approach. (caniuse.com is helpful for determining what you can use)

Reply:
Resolved by wrapping in `React.useMemo` and using for-loop for filling in optionsList.